macOS/DuckDuckGo/Freemium/DBP/FreemiumDBPFeature.swift

```diff
@@ -134,55 +146,142 @@ final class DefaultFreemiumDBPFeature: FreemiumDBPFeature {
             .sink { [weak self] _ in
                 guard let self = self else { return }
 
-                let featureAvailable = self.isAvailable
-                Logger.freemiumDBP.debug("[Freemium DBP] Subscription Updated. Feature Availability = \(featureAvailable)")
-
-                self.isAvailableSubject.send(featureAvailable)
+                Logger.freemiumDBP.debug("[Freemium DBP] Subscription Updated. Feature Availability = \(self.isAvailable)")
+                self.isAvailableSubject.send(self.isAvailable)
             }
             .store(in: &cancellables)
+
+        // Subscribe to available product updates for App Store Environment
+        if subscriptionManager.currentEnvironment.purchasePlatform == .appStore {
+            subscriptionManager.canPurchasePublisher
+                .sink { [weak self] canPurchase in
+                    guard let self = self else { return }
+
+                    // Calculate availability manually for App Store to include purchase capability
+                    // This provides more granular control over the availability calculation
+                    // compared to the general isAvailable property
+                    let featureEnabled = isEligibleFeatureFlagEnabled
+                    let notCurrentUser = isEligibleNotCurrentUser
+                    let storeFrontIsUSA = isEligibleAppStorefrontIsUSA
+                    let available = featureEnabled && notCurrentUser && storeFrontIsUSA && canPurchase
```

Comment:
This looks like a duplication of logic in `var isEligible: Bool {`

Reply:
Thanks @miasma13  - This is intentional implemented separately to `isEligible`, the reason is `isEligible` relies on the `canPurchase` property. However, we need to rely on the observed property value as at the time of observation notification the `canPurchase` property does not return the correct value. It seems this is due to some quirks of combine, where the observer is notified of the change, and passed the updated value, before the actual property value is updated.